File: backend/python_backup/verify_api.py
import urllib.request
import urllib.error
import json
import sys

BASE_URL = "http://localhost:5001"

def run_post(url, data, headers=None):
    if headers is None:
        headers = {}
    headers["Content-Type"] = "application/json"
    req_data = json.dumps(data).encode("utf-8")
    req = urllib.request.Request(f"{BASE_URL}{url}", data=req_data, headers=headers, method="POST")
    try:
        with urllib.request.urlopen(req) as res:
            return res.status, json.loads(res.read().decode("utf-8"))
    except urllib.error.HTTPError as e:
        return e.code, json.loads(e.read().decode("utf-8"))

def run_get(url, headers=None):
    if headers is None:
        headers = {}
    req = urllib.request.Request(f"{BASE_URL}{url}", headers=headers, method="GET")
    try:
        with urllib.request.urlopen(req) as res:
            return res.status, json.loads(res.read().decode("utf-8"))
    except urllib.error.HTTPError as e:
        return e.code, json.loads(e.read().decode("utf-8"))

def main():
    print("=== NextHire API Verification ===")
    
    # 1. Health check
    try:
        print("[1] Verifying server health...")
        code, health_res = run_get("/health")
        if code == 200:
            print("✓ Health: OK")
        else:
            print(f"✗ Health: Failed with code {code}")
            sys.exit(1)
    except Exception as e:
        print(f"✗ Failed to connect to server: {str(e)}")
        print("Please make sure the Flask server is running at http://localhost:5001 first.")
        sys.exit(1)
        
    # 2. Register Test User
    print("[2] Simulating User registration...")
    email = f"test_user_{int(urllib.request.time.time())}@example.com"
    reg_code, reg_res = run_post("/api/auth/register", {
        "email": email,
        "password": "testpassword123",
        "full_name": "Test Candidate"
    })
    
    if reg_code == 201:
        print(f"✓ Registration: OK. Created user: {reg_res['user']['email']}")
        token = reg_res["token"]
        headers = {"Authorization": f"Bearer {token}"}
    else:
        print(f"✗ Registration failed: {reg_res}")
        sys.exit(1)
        
    # 3. Fetch Profile
    print("[3] Fetching Profile details...")
    profile_code, profile_res = run_get("/api/auth/profile", headers=headers)
    if profile_code == 200:
        print(f"✓ Profile: OK. Credits: ₹{profile_res['credits']}")
    else:
        print(f"✗ Profile fetch failed: {profile_res}")
        sys.exit(1)
        
    # 4. Payment Order and Verification Simulator (using Credits to auto-approve)
    print("[4] Simulating Credit payment processing...")
    # Add credits to test user via SQLite directly to simulate credit purchase
    try:
        import sqlite3
        import os
        db_path = os.path.join(os.path.dirname(__file__), "instance", "nexthire.db")
        conn = sqlite3.connect(db_path)
        cursor = conn.cursor()
        cursor.execute("UPDATE users SET credits = 100 WHERE email = ?", (email,))
        conn.commit()
        conn.close()
        print("✓ Injected ₹100 credits to test user via db")
    except Exception as db_err:
        print(f"✗ Failed to inject credits to user: {db_err}")
        sys.exit(1)

    pay_code, pay_res = run_post("/api/payment/order", {
        "plan_type": "premium"
    }, headers=headers)
    
    if pay_code == 201:
        if pay_res.get('paid_with_credits'):
            print(f"✓ Payment order: OK (Fully covered by credits). Order ID: {pay_res['order_id']}")
            order_id = pay_res['order_id']
        else:
            print(f"✗ Expected order to be fully paid with credits, but got: {pay_res}")
            sys.exit(1)
    else:
        print(f"✗ Payment order creation failed: {pay_res}")
        sys.exit(1)

    # 5. Create Resume
    print("[5] Generating Resume record...")
    res_code, res_res = run_post("/api/resume", {
        "title": "Verification Resume",
        "template_id": "ats-friendly",
        "content": {
            "personal": {
                "fullName": "Test Candidate",
                "email": email,
                "phone": "+91 9999999999",
                "summary": "Full Stack developer test profile."
            },
            "skills": ["react", "python", "flask", "sql"]
        }
    }, headers=headers)
    
    if res_code == 201:
        print(f"✓ Resume: OK. Created ID: {res_res['id']}")
        resume_id = res_res["id"]
    else:
        print(f"✗ Resume creation failed: {res_res}")
        sys.exit(1)
        
    # 6. AI Summary Check
    print("[6] Verifying AI summary composer...")
    ai_code, ai_res = run_post("/api/ai/summary", {
        "profile": {
            "fullName": "Test Candidate",
            "skills": ["react", "python", "flask"],
            "experience": []
        }
    }, headers=headers)
    if ai_code == 200:
        print("✓ AI Summary: OK")
    else:
        print(f"✗ AI Summary failed: {ai_res}")
        
    # 7. Job Match recommendations
    print("[7] Auditing job recommending matcher...")
    job_code, job_res = run_get(f"/api/jobs/match/{resume_id}", headers=headers)
    if job_code == 200:
        print(f"✓ Job Matching: OK. Top matched job: {job_res[0]['job']['title']} ({job_res[0]['match_score']}% Match)")
    else:
        print(f"✗ Job Matching failed: {job_res}")
        
    print("\n✓ ALL INTEGRATION AND ROUTING TESTS PASSED!")

if __name__ == "__main__":
    main()

File: backend/verify_api.js
import mongoose from 'mongoose';
import { User } from './models/User.js';
import { config } from './config/index.js';

const BASE_URL = 'http://127.0.0.1:5001';

async function runPost(url, data, headers = {}) {
  const res = await fetch(`${BASE_URL}${url}`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      ...headers
    },
    body: JSON.stringify(data)
  });
  const text = await res.text();
  let json = {};
  try {
    json = JSON.parse(text);
  } catch (e) {
    json = { rawText: text };
  }
  return { status: res.status, data: json };
}

async function runGet(url, headers = {}) {
  const res = await fetch(`${BASE_URL}${url}`, {
    method: 'GET',
    headers: {
      ...headers
    }
  });
  const text = await res.text();
  let json = {};
  try {
    json = JSON.parse(text);
  } catch (e) {
    json = { rawText: text };
  }
  return { status: res.status, data: json };
}

async function main() {
  console.log('=== NextHire Express API Verification ===');

  // 1. Health check
  try {
    console.log('[1] Verifying server health...');
    const health = await runGet('/health');
    if (health.status === 200) {
      console.log('✓ Health: OK', health.data);
    } else {
      console.log(`✗ Health: Failed with code ${health.status}`);
      process.exit(1);
    }
  } catch (error) {
    console.error(`✗ Failed to connect to server: ${error.message}`);
    console.log('Please make sure the Express server is running at http://localhost:5001 first.');
    process.exit(1);
  }

  // 2. Register Test User
  console.log('[2] Simulating User registration...');
  const email = `test_node_user_${Date.now()}@example.com`;
  const reg = await runPost('/api/auth/register', {
    email,
    password: 'testpassword123',
    full_name: 'Node Candidate'
  });

  if (reg.status === 201) {
    console.log(`✓ Registration: OK. Created user: ${reg.data.user.email}`);
  } else {
    console.error('✗ Registration failed:', reg.data);
    process.exit(1);
  }

  const token = reg.data.token;
  const headers = { Authorization: `Bearer ${token}` };

  // 3. Fetch Profile
  console.log('[3] Fetching Profile details...');
  const profile = await runGet('/api/auth/profile', headers);
  if (profile.status === 200) {
    console.log(`✓ Profile: OK. Credits: ₹${profile.data.credits}`);
  } else {
    console.error('✗ Profile fetch failed:', profile.data);
    process.exit(1);
  }

  // 4. Verify payment order (simulated QR)
  console.log('[4] Simulating UPI QR payment order creation...');
  const pay = await runPost('/api/payment/order', {
    plan_type: 'basic',
    simulate: true
  }, headers);

  if (pay.status === 201 && pay.data.is_simulated) {
    console.log(`✓ Payment order: OK (Simulated/QR). Order ID: ${pay.data.order_id}`);
  } else {
    console.error('✗ Payment order creation failed:', pay.data);
    process.exit(1);
  }

  // 4b. Verify transaction
  console.log('[4b] Simulating UPI QR payment verification...');
  const verify = await runPost('/api/payment/verify', {
    razorpay_order_id: pay.data.order_id,
    razorpay_payment_id: `pay_qr_test_${Math.random().toString(36).substring(2, 9)}`,
    razorpay_signature: "sandbox_sig"
  }, headers);

  if (verify.status === 200) {
    console.log(`✓ Payment verification: OK. Message: ${verify.data.message}`);
  } else {
    console.error('✗ Payment verification failed:', verify.data);
    process.exit(1);
  }

  // 5. Create Resume
  console.log('[5] Generating Resume record...');
  const resume = await runPost('/api/resume', {
    title: 'Verification Resume',
    template_id: 'ats-friendly',
    content: {
      personal: {
        fullName: 'Node Candidate',
        email,
        phone: '+91 9999999999',
        summary: 'Full Stack developer test profile.'
      },
      skills: ['react', 'python', 'flask', 'sql']
    }
  }, headers);

  let resumeId = '';
  if (resume.status === 201) {
    console.log(`✓ Resume: OK. Created ID: ${resume.data.id}`);
    resumeId = resume.data.id;
  } else {
    console.error('✗ Resume creation failed:', resume.data);
    process.exit(1);
  }

  // 6. AI Summary Check
  console.log('[6] Verifying AI summary composer...');
  const aiSummary = await runPost('/api/ai/summary', {
    profile: {
      fullName: 'Node Candidate',
      skills: ['react', 'python', 'flask'],
      experience: []
    }
  }, headers);

  if (aiSummary.status === 200) {
    console.log('✓ AI Summary: OK');
  } else {
    console.error('✗ AI Summary failed:', aiSummary.data);
  }

  // 7. Job Match recommendations
  console.log('[7] Auditing job recommending matcher...');
  const jobs = await runGet(`/api/jobs/match/${resumeId}`, headers);
  if (jobs.status === 200) {
    console.log(`✓ Job Matching: OK. Top matched job: ${jobs.data[0].job.title} (${jobs.data[0].match_score}% Match)`);
  } else {
    console.error('✗ Job Matching failed:', jobs.data);
  }

  console.log('\n✓ ALL INTEGRATION AND ROUTING TESTS PASSED!');
}

main().catch(err => {
  console.error('Test runner failed:', err);
  process.exit(1);
});
